hazardous/tests/test_metrics.py

```diff
@@ -0,0 +1,301 @@
+        brier_score_incidence(
+            y_train,
+            y_test,
+            y_pred,
+            times,
+            event_of_interest="any",
+        )
+
+    with pytest.warns(None):
+        brier_score(
+            y_train,
+            y_test,
+            y_pred,
+            times,
+            event_of_interest="any",
+        )
+
+
+def test_brier_score_sampler():
+    sampler = BrierScoreSampler(y_train, random_state=0)
+    times_, y_binary, sample_weights = sampler.draw()
+
+    assert times_.shape == (y_train["event"].shape[0], 1)
+
+    y_binary_expected = np.array(
+        [0, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+        dtype=np.int32,
+    )
+
+    assert_array_equal(y_binary[:10], y_binary_expected)
+
+    sample_weights_expected = np.array(
+        [
+            1.0892326,
+            1.02340426,
+            1.0141844,
+            1.05622555,
+            1.02340426,
+            1.0892326,
+            1.0141844,
+            1.05622555,
+            1.0892326,
+            1.03891038,
+        ]
+    )
+
+    assert_array_almost_equal(sample_weights[:10], sample_weights_expected)
+
+
+def test_wrong_parameters():
+    msg = "event_of_interest must be a strictly positive integer or 'any'"
+    for event_of_interest in [-10, 0, "wrong_event"]:
+        with pytest.raises(ValueError, match=msg):
+            brier_score(
+                y_train,
+                y_test,
+                y_pred,
+                times,
+                event_of_interest,
+            )
+
+    msg = "event_of_interest must be an instance of"
+    for event_of_interest in [None, [1], (2, 3)]:
+        with pytest.raises(TypeError, match=msg):
+            brier_score(
+                y_train,
+                y_test,
+                y_pred,
+                times,
+                event_of_interest,
+            )
+
+
+def _dict_to_pd(y):
+    return pd.DataFrame(y)
+
+
+def _dict_to_recarray(y):
+    y_out = np.empty(
+        shape=y["event"].shape[0],
+        dtype=[("event", np.int32), ("duration", np.float64)],
+    )
+    y_out["event"] = y["event"]
+    y_out["duration"] = y["duration"]
+    return y_out
+
+
+@pytest.mark.parametrize("format_func", [_dict_to_pd, _dict_to_recarray])
+def test_inputs_format(format_func):
+    _, loss = brier_score(
+        format_func(y_train),
+        format_func(y_test),
+        y_pred,
+        times,
+        event_of_interest="any",
+    )
+
+    loss_expected = np.array(
+        [
+            0.01921016,
+            0.08987548,
+            0.11693115,
+            0.18832202,
+            0.21346599,
+            0.24300206,
+            0.24217776,
+            0.21987924,
+            0.19987174,
+            0.16301318,
+            0.07628881,
+            0.05829176,
+            0.0663998,
+            0.04524901,
+            0.04553689,
+            0.02250038,
+            0.02259133,
+        ]
+    )
+
+    assert_array_almost_equal(loss, loss_expected)
```

Comment:
Similar comment as above: let's not test against meaning less hardcoded values but instead against:

- either invariant mathematical properties;
- hardcoded values that directly stem from minimal toy dataset where we can derive
  the  expected values analytically (and inline comments to put the derivation).

Reply:
I agree with deriving stuff analytically, but having the results from scikit-survival allowed me to revamp the scorer and models faster. The issue is that the GPL license prevent me from importing it in the tests.

Reply:
At least make it explicit that those values were computed with scikit-survival as a reference implementation:

https://github.com/soda-inria/hazardous/pull/10#discussion_r1257849411